src/lib/kadm5/kadm_rpc.h

```diff
@@ -108,6 +108,15 @@ struct setkey3_arg {
 };
 typedef struct setkey3_arg setkey3_arg;
 
+struct setkey4_arg {
+	krb5_ui_4 api_version;
+	krb5_principal princ;
+	krb5_boolean keepold;
+	kadm5_key_data *key_data;
+	int n_key_data;
+};
```

Comment:
The API has that already built in.
If you pass a kvno of 0 the KDC will generate a new one, if you pass your own KVNO then that will be set.
I do not think a way to change the keys but retain whatever KVNO happens to be there is useful, and for those that may have some contrived reason to do something like that they can simply query the princ to find the current KVNO and then setkey using that KVNO number in the data.